server/src/main/java/com/defold/extender/services/DefoldSdkService.java

```diff
@@ -126,51 +117,112 @@ public CompletableFuture<DefoldSdk> getRemoteSdk(String hash) {
+                for (String urlPattern : configuration.getSdkUrls()) {
                     try {
-                        URL url = URI.create(String.format(url_pattern, hash)).toURL();
-
-                        ClientHttpRequestFactory clientHttpRequestFactory = new SimpleClientHttpRequestFactory();
-                        ClientHttpRequest request = clientHttpRequestFactory.createRequest(url.toURI(), HttpMethod.GET);
-
-                        // Connect and copy to file
-                        try (ClientHttpResponse response = request.execute()) {
+                        url = String.format(urlPattern, hash);
+                        URI sdkURI = URI.create(url);
+                        ClientHttpRequest existenceRequest = clientHttpRequestFactory.createRequest(sdkURI, HttpMethod.HEAD);
+                        try (ClientHttpResponse response = existenceRequest.execute()) {
                             if (response.getStatusCode() != HttpStatus.OK) {
                                 LOGGER.info("The given sdk does not exist: {} {}", url, response.getStatusCode().toString());
                                 continue;
+                            } else {
+                                // mark sdk as found for download
+                                sdkFound = true;
                             }
+                        } catch (IOException exc) {
+                            LOGGER.warn(String.format("HEAD for %s failed", url), exc);
+                        }
+                    }  catch (IOException exc) {
+                        LOGGER.warn("Can't create HEAD request", exc);
+                    }
+                }
+                if (sdkFound) {
+                    int attempt = 0;
+                    while (attempt < configuration.getMaxVerificationRetryCount()) {
+                        LOGGER.info("Downloading Defold SDK from {} attempt {} ...", url, attempt);
+                        ClientHttpRequest request;
+                        try {
+                            request = clientHttpRequestFactory.createRequest(URI.create(url), HttpMethod.GET);
+                        } catch (IOException exc) {
+                            LOGGER.error("Connect can't be established", exc);
+                            break;
+                        }
 
-                            LOGGER.info("Downloading Defold SDK from {} ...", url);
-                            Path tempDirectoryPath = Files.createTempDirectory(baseSdkDirectory, "tmp" + hash);
+                        // Connect and copy to file
+                        try (ClientHttpResponse response = request.execute()) {
+                            InputStream body = response.getBody();
+                            File tmpResponseBody = File.createTempFile(hash, ".zip.tmp");
+                            Files.copy(body, tmpResponseBody.toPath(), StandardCopyOption.REPLACE_EXISTING);
```

Comment:
Need create temporary file and write the whole response body to it because we need to use `InputStream` object twice: for checksum calculation and for unzip operation.